nibabel/bv.py

```diff
@@ -0,0 +1,691 @@
+        elif hdrDict[key]['dt'] == 'multi':
+            # empty the list in the value field
+            hdrDict[key]['value'] = []
+            # check the length of the array to expect
+            for i in range(hdrDict[hdrDict[key]['nField']]['value']):
+                # recusively iterate through the fields of all items
+                # in the array
+                hdrDict[key]['value'].append(
+                    parse_BV_header(hdrDict[key]['default'][0], fileobj))
+        else:
+            hdrDict[key]['value'] = \
+                unpack(hdrDict[key]['dt'],
+                       fileobj.read(calcsize(hdrDict[key]['dt'])))[0]
+    return hdrDict
+
+
+def pack_BV_header(hdrDict):
+    """Pack the header of a BV file format for (binary) writing to a file.
+
+    This function can be (and is) called recursively to iterate through nested
+    fields (e.g. the prts field of the VTC header).
+
+    Parameters
+    ----------
+    hdrDict: OrderedDict
+       hdrDict that contains the fields and values to for the respective
+       BV file format.
+    """
+    binaryblock = ''
+    for key in hdrDict.keys():
+        # handle zero-terminated strings
+        if hdrDict[key]['dt'] == '<s':
+            binaryblock += pack('<' + str(len(hdrDict[key]['value'])+1) + 's',
+                                hdrDict[key]['value'])
+        # handle array fields
+        elif hdrDict[key]['dt'] == 'multi':
+            # check the length of the array to expect
+            for i in range(hdrDict[hdrDict[key]['nField']]['value']):
+                # recusively iterate through the fields of all items
+                # in the array
+                binaryblock += pack_BV_header(hdrDict[key]['value'][i])
+        else:
+            binaryblock += pack(hdrDict[key]['dt'], hdrDict[key]['value'])
+    return binaryblock
+
+
+def calc_BV_header_size(hdrDict):
+    """Calculate the binary size of a hdrDict for a BV file format header.
+
+    This function can be (and is) called recursively to iterate through nested
+    fields (e.g. the prts field of the VTC header).
+
+    Parameters
+    ----------
+    hdrDict: OrderedDict
+       hdrDict that contains the fields and values to for the respective
+       BV file format.
+    """
+    hdrSize = 0
+    for key in hdrDict.keys():
+        # handle zero-terminated strings
+        if hdrDict[key]['dt'] == '<s':
+            hdrSize += calcsize('<' + str(len(hdrDict[key]['value'])+1) + 's')
+        # handle array fields
+        elif hdrDict[key]['dt'] == 'multi':
+            # check the length of the array to expect
+            for i in range(hdrDict[hdrDict[key]['nField']]['value']):
+                # recusively iterate through the fields of all items
+                # in the array
+                hdrSize += calc_BV_header_size(hdrDict[key]['value'][i])
+        else:
+            hdrSize += calcsize(hdrDict[key]['dt'])
+    return hdrSize
+
+
+class BvError(Exception):
+
```

Comment:
The PEP8 linter was OK with these blank lines?     My understanding of PEP8 is to avoid blank lines unless there is a good reason for them - for example to separate parts of a function that do different tasks.  In this case I think the blank lines aren't useful because they make the eye jump, between parts of one thing, which is a minimal class with a docstring.